platform/linux-generic/odp_packet.c

```diff
@@ -1996,7 +2016,10 @@ static inline void parse_udp(packet_parser_t *prs,
 
 	if (offset)
 		*offset   += sizeof(_odp_udphdr_t);
-	*parseptr += sizeof(_odp_udphdr_t);
+	*parseptr += _ODP_UDPHDR_LEN;
+	dport = odp_be_to_cpu_16(udp->dst_port);
+	if (dport == _ODP_UDP_VXLAN_PORT)
+		parse_vxlan(prs, in_p, parseptr, offset, frame_len, seg_len);
```

Comment:
Need to validate that the packet (and specified UDP length) is large enough to contain the VXLAN header and the encapsulated packet.